sdk/eventgrid/azure-eventgrid/azure/eventgrid/_consumer.py

```diff
@@ -20,31 +20,57 @@
 
 _LOGGER = logging.getLogger(__name__)
 
-from ._models import DeserializedEvent
+from ._models import CloudEvent, EventGridEvent
+from ._helpers import _deserialize_data
 
 class EventGridConsumer(object):
     """
     A consumer responsible for deserializing event handler messages, to allow for access to strongly typed Event objects.
     """
-
-    def deserialize_event(self, event, **kwargs):
-        # type: (Union[str, dict, bytes]) -> models.DeserializedEvent
-        """Single event following CloudEvent/EventGridEvent schema will be parsed and returned as DeserializedEvent.
-        :param event: The event to be deserialized.
-        :type event: Union[str, dict, bytes]
-        :rtype: models.DeserializedEvent
+    def decode_cloud_event(self, cloud_event, **kwargs):
```

Comment:
I know the spec in one note says decode - but i'm a little oriented towards deserialize - `deserialize_cloud_event` ... any concerns if i change it?

Reply:
Do we have precedent elsewhere in python that would dictate stronger leanings of consistency in that direction?  I'd be supportive if so; otherwise I'd be hesitant to break from what the other languages would have for such a core flow. (Putting myself as a user reading generic eg-sdk docs about "Decode" but then can't find that function here)

Reply:
You are right - there's no language precendent in python that I'm aware of. Resolving this